packages/react-core/src/hooks/__tests__/useGetUrl.test.ts

```diff
@@ -0,0 +1,195 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { waitFor } from '@testing-library/react';
+import { getUrl } from 'aws-amplify/storage';
+
+import { useGetUrl, UseGetUrlInput } from '../useGetUrl';
+
+jest.mock('aws-amplify/storage');
+
+describe('useGetUrl', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const url = 'https://amplify.s3.amazonaws.com/path/to/the/file.jpg';
+
+  const onStorageGetError = jest.fn();
+  const onGetUrlError = jest.fn();
+
+  const KEY_INPUT: UseGetUrlInput = {
+    key: 'file.jpg',
+    options: { accessLevel: 'guest' },
+    onError: onStorageGetError,
+  };
+
+  const PATH_INPUT: UseGetUrlInput = {
+    path: 'guest/file.jpg',
+    onError: onGetUrlError,
+  };
+
+  const paramType = [
+    {
+      useGetUrlParams: KEY_INPUT,
+      description: 'with key params',
+      errorHandler: 'onStorageGetError',
+    },
+    {
+      useGetUrlParams: PATH_INPUT,
+      description: 'with path params',
+      errorHandler: 'onGetUrlError',
+    },
+  ];
+
+  it.each(paramType)(
+    `should return true for isLoading at initialization $description`,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    async ({ useGetUrlParams, description }) => {
+      (getUrl as jest.Mock).mockResolvedValue({ url });
+      const { result } = renderHook(() => useGetUrl(useGetUrlParams));
+      await waitFor(() => {
```

Comment:
Can we replace `waitFor` with `waitForNextUpdate`? Prefer not using multiple test libraries unless actually required